src/Microsoft.Windows.CsWin32/Generator.cs

```diff
@@ -2874,6 +2874,15 @@ private StructDeclarationSyntax DeclareTypeDefStruct(TypeDefinition typeDef)
                 .WithExpressionBody(ArrowExpressionClause(AssignmentExpression(SyntaxKind.SimpleAssignmentExpression, fieldAccessExpression, valueParameter)))
                 .WithSemicolonToken(Token(SyntaxKind.SemicolonToken)));
 
+            if (fieldInfo.FieldType is IdentifierNameSyntax { Identifier: { Value: "nint" or "nuint" or nameof(IntPtr) or nameof(UIntPtr) } })
```

Comment:
Null is not a value for an integer. Zero is. Can we drop `nint` and `nuint` checks from this?
Or if you need it for those, what if we call it `IsZero`? (at least for the non-pointer field types)

Reply:
And I'm doubtful about the value of IsZero, since I suspect `== 0` is legal syntax for these structs.

Reply:
@AArnott HGDIOBJ has a `nint` field. I don't care at all about if `nint` works per say, but I absolutely want handles to have `IsNull`.

Reply:
I'd say the issue then is that `HGDIOBJ` uses `nint` when it should be using `IntPtr`. We may be able to work this out without hard-coding when we address #199, but until then I've hard-coded it so that your test can pass without adding `IsNull` to all `nint` structs.